node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/provider/ComponentsProviderImpl.java

```diff
@@ -24,63 +25,52 @@
 import com.yahoo.vespa.hosted.node.admin.orchestrator.OrchestratorImpl;
 import com.yahoo.vespa.hosted.node.admin.util.ConfigServerHttpRequestExecutor;
 import com.yahoo.vespa.hosted.node.admin.util.Environment;
-import com.yahoo.vespa.hosted.node.admin.util.SecretAgentScheduleMaker;
 
-import java.io.IOException;
-import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.time.Clock;
-import java.util.Set;
+import java.time.Duration;
 import java.util.function.Function;
 
+import static com.yahoo.vespa.defaults.Defaults.getDefaults;
+
 /**
  * Set up node admin for production.
  *
  * @author dybis
  */
-public class ComponentsProviderImpl implements ComponentsProvider {
-    private static final ContainerName NODE_ADMIN_CONTAINER_NAME = new ContainerName("node-admin");
-
-    private final NodeAdminStateUpdater nodeAdminStateUpdater;
-    private final MetricReceiverWrapper metricReceiverWrapper;
-
-    private static final int NODE_AGENT_SCAN_INTERVAL_MILLIS = 30000;
+public class ComponentsProviderImpl extends AbstractComponent implements ComponentsProvider {
     private static final int WEB_SERVICE_PORT = getDefaults().vespaWebServicePort();
+    private static final Duration NODE_AGENT_SCAN_INTERVAL = Duration.ofSeconds(30);
+    private static final Duration NODE_ADMIN_CONVERGE_STATE_INTERVAL = Duration.ofSeconds(30);
 
-    // Converge towards desired node admin state every 30 seconds
-    private static final int NODE_ADMIN_CONVERGE_STATE_INTERVAL_MILLIS = 30000;
+    private final NodeAdminStateUpdater nodeAdminStateUpdater;
+    private final Lock classLock;
 
     @Inject
-    public ComponentsProviderImpl(Docker docker, MetricReceiverWrapper metricReceiver) {
-        String baseHostName = HostName.getLocalhost();
-        Environment environment = new Environment();
-        Set<String> configServerHosts = environment.getConfigServerHosts();
-        if (configServerHosts.isEmpty()) {
-            throw new IllegalStateException("Environment setting for config servers missing or empty.");
-        }
+    public ComponentsProviderImpl(Docker docker, MetricReceiverWrapper metricReceiver, Locking locking) {
+        classLock = locking.lock(this.getClass());
```

Comment:
This should presumably start a try-finally block?

Reply:
Surrounded with regular try-catch.